File: Programmers/[12907] 거스름돈/ONE/12907.java
class Solution {
    public int solution(int n, int[] money) {

        int[] dp = new int[n + 1];
        dp[0] = 1;

        for (int m : money) {
            for (int change = 1; change <= n; change++) {
                if (change >= m) {
                    dp[change] += dp[change - m];
                }
            }
        }
        return dp[n];
    }
}

File: Programmers/[12907] 거스름돈/KingDonggyu/12907.js
const DIVIDER = 1000000007;

function solution(n, money) {
  const dp = Array(n + 1).fill(0);

  dp[0] = 1;

  money.forEach((m) => {
    for (let change = 0; change <= n; change++) {
      if (change >= m) {
        dp[change] += dp[change - m] % DIVIDER;
      }
    }
  });

  return dp[n];
}
